README.md

```diff
@@ -1,4 +1,114 @@
 QueueNado
 =========
 
-Implementation of fast queue patterns (Push/Pull, request/reply, pub/sub, dealer/router) for C++
+`QueueNado` is a collection of C++ implemntations of ZMQ fast queue patterns (Push/Pull, request/reply, pub/sub, dealer/router) for C++
+
+
+# Rifle - Vampire
+`Rifle - Vampire` implements the [push / pull](http://zguide.zeromq.org/page:all#Divide-and-Conquer) messaging pattern in zmq. The communication setup supports the following client/server communications
+
+**Client -to- Server**
+* one-to-one
+* one-to-many
+* many-to-one
+
+In a *many* scenario, it is the "one" side that has to do ```bind``` ([[Vampire.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Vampire.h) [[Rifle.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Rifle.h) on the queue.
+#### Rifle - Vampire Limitation
+It **cannot** do many-to-many
+
+#### Use Cases for `Rifle - Vampire`
+* Reliable messaging without responses
+* High performance (500k msgs per second or higher) with zero_copy
+* Process to process communication
+
+#### API
+[[Vampire.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Vampire.h)
+[[Rifle.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Rifle.h)
+
+#### Test usage
+[[RifleVampireTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/RifleVampireTests.cpp)
+
+# Shotgun - Alien
+`Shotgun - Alien` implements the [pub / sub](http://zguide.zeromq.org/page:all#Getting-the-Message-Out) messaging pattern in zmq.
+
+#### Use Cases for `Shotgun - Alien`
+* One to many: one sender communicating with many listeners.
+* Not high performance around 10k msgs a sec. This can be improved by batching many messages together.
+* Process to process communication
+
+#### Known limitations and issues
+* [Slow joiner](http://zguide.zeromq.org/php:chapter5#Representing-State-as-Key-Value-Pairs) issues don't matter or can be worked around
+* All listeners need to see every message for a topic
+
+#### API
+[[Shotgun.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Shotgun.h)
+[[Alien.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Alien.h)
+
+#### Test usage
+[[ShotgunAlienTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/ShotgunAlienTests.cpp)
+
+
+# Headcrab - Crowbar
+`Headcrab - Crowbar` implements [request / reply](http://zguide.zeromq.org/page:all#Ask-and-Ye-Shall-Receive) messaging pattern in zmq.
+
+#### Use Cases for `Headcrab - Crowbar` 
+* Need reply to guarantee message was delivered
+* Know that their will always be a response (otherwise the socket will be broken)
```

Comment:
Rewrite suggestion: Need to receive an acknowledgement from the receiver for each message sent to it.